apps/shipments/iot_client.py

```diff
@@ -68,10 +69,13 @@ def get_list_owner_devices(self, owner_id, active=None, in_bbox=None):
                           f' params: {params_dict}. Error message: {list_devices["error"]}')
                 raise AWSIoTError(f'Error in AWS IoT response: {list_devices["error"]}')
 
-            new_devices = list_devices['data'].get('devices', None)
+            new_devices = list_devices['data'].get('devices')
             if new_devices:
-                results.extend(new_devices)
+                for device in new_devices:
+                    for cert_key_name in ('certificate_id', 'certificateId', ):
+                        device = remove_dict_key_recursively(device, cert_key_name)
```

Comment:
Instead of calling `remove_dict_key_recursively` twice (recursive functions are expensive), it makes more sense to modify `remove_dict_key_recursively` to accept a set/tuple of multiple keys to be removed.